cli/server.go

```diff
@@ -1219,8 +1245,19 @@ func configureTLS(tlsMinVersion, tlsClientAuth string, tlsCertFiles, tlsKeyFiles
 	if err != nil {
 		return nil, xerrors.Errorf("load certificates: %w", err)
 	}
+	var selfSignedCertificate *tls.Certificate
+	if len(certs) == 0 {
+		selfSignedCertificate, err = generateSelfSignedCertificate()
+		if err != nil {
+			return nil, xerrors.Errorf("generate self signed certificate: %w", err)
+		}
```

Comment:
Why not just `certs = append(certs, selfSignedCertificate)` instead of the extra logic?